src/util.ts

```diff
@@ -1,8 +1,25 @@
+/** Limits an array to a given length.
+ *
+ * @param array - The array to limit.
+ * @param limit - The maximum length the array can have.
+ */
 function limitArray(array: any[], limit?: number): any[] {
   if (limit && (array.length > limit)) {
     return array.slice(array.length - limit, array.length);
   }
   return array;
 }
+/** Filters the given array with an async function
+ *
+ * @param array - The array to filter.
+ * @param filterfn - The function to filter the array with.
+ */
+async function filterAsync(array: any[], filterfn: (value: any) => any): Promise<any[]> {
```

Comment:
It's a good idea to avoid using the `any` type. It kind of beats the point of using typescript. Instead, you can do something like this:

``` typescript
export function mapAsync<T, U>(array: T[],
    callbackfn: (value: T, index: number, array: T[]) => Promise<U>): Promise<U[]> {
    return Promise.all(array.map(callbackfn));
}

export async function filterAsync<T>(array: T[],
    callbackfn: (value: T, index: number, array: T[]) => Promise<boolean>): Promise<T[]> {
    const filterMap = await mapAsync(array, callbackfn);
    return array.filter((value, index) => filterMap[index]);
}
```